precompiles/utils/src/solidity/codec/bytes.rs

```diff
@@ -107,6 +109,8 @@ impl<K: Kind, S: Get<u32>> Codec for BoundedBytesString<K, S> {
 			.get(range)
 			.ok_or_else(|| RevertReason::read_out_of_bounds(K::signature()))?;
 
+		reader.move_cursor(array_size)?;
```

Comment:
I'm pretty sure this one is wrong, see my other comment for more explanation. We don't want to advance over the referenced data, as the cursor's position needs to be at the next element in the tightly-packed value area.

I think the test case you added misses this detail because it only has one reference type; if you had multiple strings it would catch this.